### PR TITLE
Ce/OIDC

### PR DESCRIPTION
**Pull request by calellowitz**
This PR sets up the OAuth provider and configures a few extra fields to be returned by the openid connect userinfo view.

### CODE REVIEW

File: users/views.py [hunk 1]
```diff
@@ -150,17 +146,18 @@ def recover_secondary_phone(request):
         return HttpResponse(status=401)
     if status.step != RecoveryStatus.RecoverySteps.CONFIRM_SECONDARY:
         return HttpResponse(status=401)
-    device = PhoneDevice.objects.get(phone_number=user.recovery_phone, user=user)
-    device.generate_challenge()
+    otp_device, _ = PhoneDevice.objects.get_or_create(phone_number=user.recovery_phone, user=user)
+    otp_device.save()
```
Comment (snopoke) [inline users/views.py]: what's the extra `save` call for?

File: users/oauth.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from oauth2_provider.oauth2_validators import OAuth2Validator
+
+class ConnectOAuth2Validator(OAuth2Validator):
+
+    def get_additional_claims(self, request):
+        claims = {}
+        claims["sub"] = request.user.username
```
Comment (snopoke) [inline users/oauth.py:7]: what is `sub`?

File: users/views.py (hunk 1 above)
Comment (calellowitz) [reply users/views.py]: I think this is a weird merge artifact. I did not change that view, and it's not in the full diff.

File: users/oauth.py (hunk 2 above)
Comment (calellowitz) [reply users/oauth.py:7]: It stands for "subscriber." It is the normal oauth field for the user identifier (in theory we could use anything but it seemed more correct to just modify this field). It defaults in django-oauth-toolkit to the primary key in the user table, but username seemed more reasonable. You can see it used in the SSO PR when we are saving the link.

**Review by snopoke — APPROVED** [review]

**Pull request merged**